### ISSUE TITLE
Adding bot to active directory

### ISSUE THREAD

**Issue by 01** [MicrosoftDocs/msteams-docs#611]
To achieve proactive messaging (starting conversation) on teams in a single tenancy without having user first message the bot are there any work arounds? Is it possible to add bot to active directory and for all intents and purpose be a user and can be messaged and message others? 

Is the unique user id generated by the Bot Adapter or is it generated by Teams as part of the message post to the bot. If its the former would it be possible to pass on a request  to a single bot to other adapters to get the unique user id for other bots, achieving ability to create conversations proactively for all bots within a tenancy by first messaging one bot? 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ba8d3979-14f5-3105-ad89-ce6f98d11df1
* Version Independent ID: e53a7d1e-0e70-ca30-9809-fda7ebe5f4da
* Content: [Proactive messages - Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bot-conversations/bots-conv-proactive)
* Content Source: [msteams-platform/concepts/bots/bot-conversations/bots-conv-proactive.md](https://github.com/MicrosoftDocs/msteams-docs/blob/master/msteams-platform/concepts/bots/bot-conversations/bots-conv-proactive.md)
* Product: **msteams**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

**Comment by Wajeed-msft** [comment]
1. Currently, only workaround is to [create a Team with all users](https://github.com/MicrosoftDocs/msteams-docs/issues/464#issuecomment-488585187).
2. [User's uniqueId](https://docs.microsoft.com/en-us/microsoftteams/platform/concepts/bots/bot-conversations/bots-conv-proactive#starting-personal-conversations) differs from Bot to Bot. You can not use same id to send proactive message from another bot.

**Comment by 01** [comment]
@Wajeed-msft  I understand number 2. Let me rephrase the question, how is that uniqueId generated? If my organization has multiple bots could I design a middleware that passes the message activity to the other bots to get the user ids effectively "registering" a  userid with all the bots once it any single bot is messaged for the first time

**Comment by hradawski** [comment]
Exactly! Good Question is, how to create, wich is the algoritm to create those TeamsUsersId isolated.

**Comment by billbliss** [comment]
The algorithm is a one-way hash for privacy reasons, so we have no plans to expose it.
